toolparameteriser/testcreator.py

```diff
@@ -98,37 +132,49 @@ def _get_test_parameters(self):
                 exit()
     
     def __prepare_run_dir(self,runID:str,params:dict) -> str:
+
+        logging.info("Preparing output directory for {id}.".format(id=runID))
         outpath=os.path.join(self.Config["Output_path"],runID)
         os.makedirs(outpath)
+        logging.debug("Output directory, {p}, created successfully.".format(p=outpath))
 
         # testing if user has specified 
         try:
             allinputfiles = glob.glob(self.Config['input']['path'], recursive=False)
         except KeyError:
             # below commented statemnt should be placed in a "pre-screening" function
-            # logging.info('"Input" not specified in config toml file. Not copying files to output directory.')
+            logging.info('"Input" not specified in config toml file. Not copying files to output directory.')
             return outpath
 
-        try:
-            if "numfiles" in params.keys(): 
-                runfiles = random.sample(allinputfiles, k=int(params["numfiles"]))
-            else:
-                runfiles = random.sample(allinputfiles, k=int(self.Config["jobs"]["numfiles"]))
-        except KeyError:
-            logging.fatal('"numfiles" parameter not provided in either config TOML or profiles CSV files.')
-            raise KeyError
+        if "numfiles" in params.keys(): 
+            numfiles = int(params["numfiles"])
+        elif "numfiles" in self.Config["jobs"].keys():
+            numfiles = int(self.Config["jobs"]["numfiles"])
+        else:
+            logging.fatal('"numfiles" parameter not supplied in either the config or job parameters files.')
+            exit()
```

Comment:
Throw errors instead of using `exit()` in case this is used as a library by another python package

Reply:
Acknowledging your comment, but raising that exit() is used throughout. I'll make the change a later.

Reply:
Created issue https://github.com/WEHI-ResearchComputing/ToolParameteriser/issues/17